ydb/core/kqp/host/kqp_gateway_proxy.cpp

```diff
@@ -908,15 +908,133 @@ class TKqpGatewayProxy : public IKikimrGateway {
     }
 
     TFuture<TGenericResult> CreateGroup(const TString& cluster, const TCreateGroupSettings& settings) override {
-        FORWARD_ENSURE_NO_PREPARE(CreateGroup, cluster, settings);
+        CHECK_PREPARED_DDL(CreateGroup);
+
+        auto createGroupPromise = NewPromise<TGenericResult>();
+
+        TString database;
+        if (!SetDatabaseForLoginOperation(database, GetDomainLoginOnly(), GetDomainName(), GetDatabase())) {
+            return MakeFuture(ResultFromError<TGenericResult>("Couldn't get domain name"));
+        }
+
+        NKikimrSchemeOp::TModifyScheme schemeTx;
```

Comment:
With these changes we get a new code that does the same things as in `kqp_ic_gateway.cpp`. Can we make some library with helpers that fill `TModifyScheme` proto and is used by both gateways?

Reply:
It is not a strict requirement. I think we need to see if it will bring us a profit. Maybe in this particular cases protos are very simple. If so, maybe it would be better to leave this code in both gateways.